design/tracing-design.md

```diff
@@ -0,0 +1,170 @@
+# Design for Supporting Tracing in Contour
+
+Status: Draft
+
+## Abstract
+Envoy has rich support for [distributed tracing][1].
+However, Contour does not currently offer a way to configure Envoy for tracing.Exporting trace data to OpenTelemetry has been supported since envoy v1.23. This design document provides the implementation of exporting tracing data to [OpenTelemetry][2] in Contour.
+
+## Background
+Envoy's [documentation on tracing][1] says:
+> Distributed tracing allows developers to obtain visualizations of call flows in large service oriented architectures. It can be invaluable in understanding serialization, parallelism, and sources of latency.
+ 
+Envoy supports:
+- generating traces for incoming requests
+- joining existing traces when a client sets the `x-client-trace-id` header
```

Comment:
Is this header allowed by default?

Reply:
yes，It is allowed by default.
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/runtime#config-http-conn-man-runtime-client-enabled